app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/InputDefinition.scala

```diff
@@ -1,17 +1,51 @@
 package com.azavea.rf.datamodel
 
-import com.azavea.rf.datamodel._
-import io.circe.generic.JsonCodec
-
-import geotrellis.vector._
-import geotrellis.vector.io._
-
 import java.util.UUID
 
-@JsonCodec
+import cats.implicits._
+import com.azavea.rf.tool.ast.MapAlgebraAST
+import com.azavea.rf.tool.params.EvalParams
+import geotrellis.vector.MultiPolygon
+import io.circe._
+import io.circe.generic.JsonCodec
+import io.circe.syntax._
+
+/** 2017 May 19 @ 13:30
+  * There are two varieties Export - those that involve an AST, which
+  * performs Map Algebra and produces [[SinglebandGeoTiff]]s, and as-ingested
+  * [[MultibandTile]] exports which can perform colour correction and do simple
+  * cropping / stitching.
+  */
 case class InputDefinition(
-  projectId: UUID,
+  projectId: UUID,  // TODO: Might not be necessary.
   resolution: Int,
-  layers: Array[ExportLayerDefinition],
-  mask: Option[MultiPolygon]
+  style: Either[SimpleInput, ASTInput]
 )
+
+object InputDefinition {
+  implicit val dec: Decoder[InputDefinition] = Decoder.instance(c =>
+    (c.downField("projectId").as[UUID]
+      |@| c.downField("resolution").as[Int]
+      |@| c.downField("style").as[SimpleInput].map(Left(_))
+           .orElse(c.downField("style").as[ASTInput].map(Right(_)))
+    ).map(InputDefinition.apply)
+  )
+
+  implicit val eitherEnc: Encoder[Either[SimpleInput, ASTInput]] = new Encoder[Either[SimpleInput, ASTInput]] {
+    final def apply(a: Either[SimpleInput, ASTInput]): Json = a match {
+      case Left(l) => l.asJson
+      case Right(r) => r.asJson
+    }
+  }
```

Comment:
This could be more generic, but I think there is already such an instance built-in in circe which gives output like:
```js
{"Left": <your object>}
```
which is not desirable.